src/core/eval.rs

```diff
@@ -325,7 +327,7 @@ impl DAG {
           let Ref { nam, exp, ast, parents: ref_parents, .. } =
             unsafe { &mut *link.as_ptr() };
           if let Some(def) = defs.get(exp) {
-            let parents = ref_parents.clone();
+            let parents = *ref_parents;
```

Comment:
In this case `ref_parents` implements `Copy` if I understand correctly, so it is more efficient to use dereferencing, since `clone` defers to it?
https://rust-lang.github.io/rust-clippy/master/#clone_on_copy
I could be wrong here though, I don't remember exactly.

Reply:
Oh, you're right, `ref_parents` is just a `&` pointer to a `NonNull` pointer to the DLL, so yeah, I think cloning is totally fine. In my previous comment I thought we could be invoking a Clone on the DLL itself (although now that I look I think we don't actually implement that trait) , so wanted to be cautious